cpp/order/OrderParameter.h

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2010-2019 The Regents of the University of Michigan
+// This file is from the freud project, released under the BSD 3-Clause License.
+
+#ifndef ORDER_PARAMETER_H
+#define ORDER_PARAMETER_H
```

Comment:
We need to rename this to something less generic. As we've established, good naming is hard, and I don't want to waste your timing on coming up with good names, so for now I'm OK with literally naming the parent class HexTransOrderParameter or something equally banal and just getting things working.